src/Parsers/ParserDescribeTableQuery.cpp

```diff
@@ -19,17 +21,23 @@ bool ParserDescribeTableQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & ex
     ParserKeyword s_settings(Keyword::SETTINGS);
     ParserSetQuery parser_settings(true);
 
-    ASTPtr database;
-    ASTPtr table;
+    ASTPtr select;
 
     if (!s_describe.ignore(pos, expected) && !s_desc.ignore(pos, expected))
         return false;
 
     auto query = std::make_shared<ASTDescribeQuery>();
-
     s_table.ignore(pos, expected);
 
-    if (!ParserTableExpression().parse(pos, query->table_expression, expected))
+    if (ParserSelectWithUnionQuery().parse(pos, select, expected))
+    {
+        auto table_expr = std::make_shared<ASTTableExpression>();
+        table_expr->subquery = select;
+        table_expr->children.push_back(select);
```

Comment:
The AST fuzzer is crashing, maybe because of this. Shouldn't the SELECT be wrapped with an `ASTWithAlias`?